File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    char *text = get_string("Text: ");

    int letters = count_letters(text);
    printf("%i\n", letters);
    int words = count_words(text);
    printf("%i\n", words);
    int sentences = count_sentences(text);
    printf("%i\n", sentences);
    float fl_words = words;

    int L = (letters / fl_words) * 100;
    printf("%i\n", L);
    int S = (sentences / fl_words) * 100;
    printf("%i\n", S);
    int grade = round(0.0588 * L - 0.296 * S - 15.8);

    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }

    else if (grade >= 16)
    {
        printf("Grade 16+\n");
    }

    else
    {
        printf("Grade %i\n", grade);
    }
}

int count_letters(string text)
{
    int letters = 0;

    for (int i = 0; text[i] != '\0'; i++)
    {
        if (isalpha(text[i]))
        {
            letters++;
        }
    }
    return letters;
}

int count_words(string text)
{
    int words = 1;

    for (int i = 0; i < strlen(text); i++)
    {
        if (text [i] == ' ')
        {
            words++;
        }
    }
    return words;
}

int count_sentences(string text)
{
    int sentences = 0;

    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            sentences++;
        }
    }
    return sentences;
}

File: sentimental-readability/readability.py
from cs50 import get_string

word_counter = 1
sentence_counter = 0
letter_counter = 0

text = get_string("Text: ")

for c in text:
    if c == " ":
        word_counter += 1
    elif c in [".", "!", "?"]:
        sentence_counter += 1
    elif c in ["'", ",", ":", ";"]:
        continue
    else:
        letter_counter += 1

L = (letter_counter / word_counter) * 100
S = (sentence_counter / word_counter) * 100

grade = round(0.0588 * L - 0.296 * S - 15.8)

if grade >= 16:
    print("Grade 16+")
elif grade < 1:
    print("Before Grade 1")
else:
    print(f"Grade: {grade}")

# print(word_counter)
# print(sentence_counter)
# print(letter_counter)
# print(grade)